e2e/mapimachines_test.go

```diff
@@ -0,0 +1,171 @@
+					break
+				}
+			}
+			if !foundMapiMachine {
+				Skip("No machine found with AuthoritativeAPI set to MachineAPI")
+			}
+			// Define the new machine based on the template.
+			newMachine = &machinev1.Machine{
+				ObjectMeta: metav1.ObjectMeta{
+					// Using a predictable name based on the testcase in polarion.
+					Name:      fmt.Sprintf("%s-test-%s", templateMachine.Name, "ocp-81829"),
+					Namespace: templateMachine.Namespace,
+					Labels:    templateMachine.Labels,
+				},
+				Spec: *templateMachine.Spec.DeepCopy(),
+			}
+			// Clear status and other instance-specific fields that should not be copied.
+			newMachine.Spec.ProviderID = nil
+			newMachine.ObjectMeta.Labels = nil
+			By(fmt.Sprintf("Creating a new machine in namespace: %s", newMachine.Namespace))
+			// Create the new machine object in the cluster.
+			err = cl.Create(ctx, newMachine)
+			Expect(err).NotTo(HaveOccurred(), "Failed to create new machine")
+			By("Waiting for the new machine to enter 'Running' phase")
+			Eventually(komega.Object(newMachine), DefaultTimeout, DefaultInterval).
+				Should(HaveField("Status.Phase", HaveValue(Equal("Running"))),
+					fmt.Sprintf("Machine %s did not enter 'Running' phase within the timeout", newMachine.Name))
+
+			By("Verifying the new machine has a node reference")
+			Eventually(komega.Object(newMachine)).Should(HaveField("Status.NodeRef", Not(BeNil())))
+
+			By("Verifying status of mapi machine for synchronisation done successfully")
+			Expect(newMachine.Status.Conditions).To(ContainElement(
+				SatisfyAll(
+					HaveField("Type", ConditionSynchronized),
+					HaveField("Status", corev1.ConditionTrue),
+					HaveField("Message", ContainSubstring("Successfully synchronized MAPI Machine to CAPI")),
+				),
+			))
+
+			Expect(newMachine.Status.Conditions).To(ContainElement(
+				SatisfyAll(
+					HaveField("Type", MapiConditionPaused),
+					HaveField("Status", corev1.ConditionFalse),
+					HaveField("Message", ContainSubstring("The AuthoritativeAPI is set to MachineAPI")),
+				),
+			))
+
+			By("Verifying the corresponding machine in openshift-cluster-api is paused")
+			// Get all CAPI machines.
+			allMachines, err := framework.GetMachines(cl)
+			Expect(err).NotTo(HaveOccurred(), "Failed to get CAPI machines")
+			// Find the CAPI machine that corresponds to our new MAPI machine by name.
+			var targetMachine *clusterv1.Machine
+			for _, capiMachine := range allMachines {
+				if capiMachine.Name == newMachine.Name {
+					targetMachine = capiMachine
+					break
+				}
+			}
+			pausedFound := false
+			for _, cond := range targetMachine.Status.V1Beta2.Conditions {
+				if cond.Type == CapiConditionPaused &&
+					string(cond.Status) == "True" &&
+					strings.Contains(cond.Message, "cluster.x-k8s.io/paused annotation") {
+					pausedFound = true
+					break
+				}
+			}
+			Expect(pausedFound).To(BeTrue(), "Expected Paused condition with Status=True and correct Message substring")
```

Comment:
This could be done directly in ginkgo and then it would print out the full conditions if it failed

```
Expect(targetMachine.Status.V1Beta2.Conditions).To(ContainElement(SatisfyAll(
  HasField("Type", Equal(CapiConfitionPaused)),
  HasField("Status", Equal("True")),
  HasField("Message", ContainSubstring("cluster.x-k8s.io/paused annotation")),
)))

```